### PR TITLE
add EB option --generate-devel-module

### PR DESCRIPTION
**Pull request by smoors**
this option is useful for example when creating a module in combination with `--module-only` and `-modules-footer` where the footer generates an `LmodError` upon loading the module.

using `LmodError` causes EB to fail because it tries to load the module unless make_devel_module() is skipped. skipping this function is fine in this case because the module is not supposed to be loaded anyway.

if this option is well-received I'll try to write a test for it as well.

### CODE REVIEW

**Comment by smoors** [comment]
closing-reopening

**Comment by boegel** [comment]
@smoors Can you clarify the use case for this? What's the point of generating a module file you can't load?

Also, this is for using together with `--force --module-only`, right? Since otherwise the generated module will be loaded during the sanity check, which will result in a failure if the module can't be loaded.

I would rename it too, since it's a boolean option. It should be something like `--generate-devel-module` that is enabled by default, and then you can use `--disable-generate-devel-module`.

That being said, I think we need to re-evaluate generating those devel modules, I doubt they're actually useful at all (see also #109).

**Comment by smoors** [comment]
@boegel my use case is discoverability:
you could make a module visible to the user e.g. when they do 'module spider' or 'module av' on the login node, without allowing them to actually load/use it on the login node itself. when the user tries to load it they then get an error message saying 'this module can only be used on this type of node'. currently we work around this by creating a `LmodWarning` but imho this is confusing for the users because then they have loaded a module that does not do anything, while still having loaded its dependencies.

it should indeed be used together with `--force`.

renaming the is fine to me, I was not happy about the name in the first place.

about the usefulness of devel modules, I never used them myself, if they work at all. maybe we can ask the community to see if anyone is using them?

**Comment by smoors** [comment]
closing-reopening

File: easybuild/tools/options.py [hunk 1]
```diff
@@ -1063,6 +1065,10 @@ def _postprocess_config(self):
         if self.options.inject_checksums:
             self.options.pre_create_installdir = False
 
+        # imply --force with --disable-generate-devel-module
+        if not self.options.generate_devel_module:
+            self.options.force = True
```
Comment (boegel) [inline easybuild/tools/options.py:1070]: @smoors This seems weird to me, I don't think `--disable-generate-devel-module` should always imply `--force`?

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/tools/options.py (hunk 1 above)
Comment (smoors) [reply easybuild/tools/options.py:1070]: yeah I was not sure about that, I'll remove it.

**Review by boegel — APPROVED** [review]

**Pull request merged**